### PR TITLE
platform: ace: Set fastest clock on platform init

### PR DESCRIPTION
**Pull request by kfrydryx**
Switch to fast clock on init using zephyr clock controller.

### CODE REVIEW

**Review by kv2019i — APPROVED** [review]
Excellent, thanks @kfrydryx !

File: src/platform/intel/ace/lib/clk.c [hunk 1]
```diff
@@ -11,9 +11,26 @@
...
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clk, LOG_LEVEL_DBG);
 
 static SHARED_DATA struct clock_info platform_clocks_info[NUM_CLOCKS];
 
+static int select_cpu_freq(int clock, int hz)
+{
+	const struct device *clock_ctrl = device_get_binding("clkctl");
```
Comment (juimonen) [inline src/platform/intel/ace/lib/clk.c:23]: I think the get_binding was deprecated etc. so you'd probably want to use something like in ace/lib/dma.c:
DEVICE_DT_GET(DT_NODELABEL(foobar))

File: src/platform/intel/ace/lib/clk.c [hunk 2]
```diff
@@ -11,9 +11,26 @@
...
+
+	if (!clock_ctrl) {
+		LOG_ERR("Could not find clock controller!");
+		return -ENODEV;
+	}
```
Comment (juimonen) [inline src/platform/intel/ace/lib/clk.c:28]: and as it is compile time assignment you probably dont need the checks

File: src/platform/intel/ace/lib/clk.c (hunk 1 above)
Comment (kfrydryx) [reply src/platform/intel/ace/lib/clk.c]: Right. There's no info about it in the docs for `device_get_binding`, but it's there in zephyrs release notes.

File: src/platform/intel/ace/lib/clk.c (hunk 2 above)
Comment (kfrydryx) [reply src/platform/intel/ace/lib/clk.c]: removed

**Comment by juimonen** [comment]
static int select_cpu_freq(int clock, int hz)
{
        return clock_control_set_rate(DEVICE_DT_GET(DT_NODELABEL(clkctl)), NULL,
                               (clock_control_subsys_rate_t)hz);
}

you could even do the above... not sure what the set_rate is returning

**Comment by aiChaoSONG** [comment]
> you could even do the above... not sure what the set_rate is returning

 * @retval -EALREADY if clock was already in the given rate.
 * @retval -ENOTSUP If the requested mode of operation is not supported.
 * @retval -ENOSYS if the interface is not implemented.
 * @retval other negative errno on vendor specific error.

**Review by abonislawski — APPROVED** [review]

File: src/platform/intel/ace/lib/clk.c [hunk 3]
```diff
@@ -11,9 +11,18 @@
 #include <sof/common.h>
 #include <sof/lib/memory.h>
 #include <rtos/spinlock.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
```
Comment (abonislawski) [inline src/platform/intel/ace/lib/clk.c:16]: no longer needed

Comment (kfrydryx) [reply src/platform/intel/ace/lib/clk.c]: ok, removed

**Review by juimonen — APPROVED** [review]
thanks! lgtm.

**Review by lgirdwood — APPROVED** [review]

**Review by marcinszkudlinski — APPROVED** [review]
please merge ASAP

**Comment by kv2019i** [comment]
@wszypelt @kfrydryx can you check the Intel CI results? I think we can ignore "pr-device-test" as the tests don't cover the hw this PR modifies, so they should be good.

**Review by tmleman — APPROVED** [review]

**Comment by kv2019i** [comment]
The Intel CI check still fails, "14:01:04,382 ERROR - Step 'master>SCANS>TESTS>TRIGGER PYTHON SMOKE TESTS' is failed: Step is failed since the triggered build is failed, cancelled, or timed out." 

@kfrydryx @wszypelt Can we ignore this or can this be resolved?

**Comment by abonislawski** [comment]
@kv2019i we shouldnt ignore Internal CI, this is ofc CI issue but when resolved it will run our py tests on hw so they are important to pass
trying to rerun now

**Review by aiChaoSONG — APPROVED** [review]

**Comment by kv2019i** [comment]
All good now, merging.

**Pull request merged**